Task3/Task3/Task3/Boxes/Box.cs

```diff
@@ -0,0 +1,163 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Task3.Colors;
+using Task3.Exceptions;
+using Task3.Figures;
+using Task3.FileManagment;
+using Task3.Infrastructure;
+
+namespace Task3.Boxes
+{
+    public class Box
+    {
+        public List<Figure> BoxOfFigures = new List<Figure>(20);
+        
+        private readonly IBoxFileWorker _boxFileWorker;
+
+        public Box()
+        {
+
+        }
+
+        public Box(IBoxFileWorker boxFileWorker)
+        {
+            _boxFileWorker = boxFileWorker;
+        }
+
+        public void AddFigure(Figure item)
+        {
+            var alreadyExist = true;
+
+            foreach (var figure in BoxOfFigures)
+            {
+                if (item.GetType() == figure.GetType() && item.Side == figure.Side)
+                {
+                    alreadyExist = BoxOfFigures.Contains(item);
+                }
+            }
+
+            if (alreadyExist == false)
+            {
+                throw new AddException("This figure is already in the box");
+            }
```

Comment:
I don't understand this logic.
Why do we set already exiast to true?

Reply:
fixed